platform/wayland/cog-platform-wl.c

```diff
@@ -1373,24 +1373,29 @@ cog_wl_platform_create_im_context(CogPlatform *platform)
 }
 
 void
-cog_wl_platform_popup_create(CogWlPlatform *platform, WebKitOptionMenu *option_menu)
+cog_wl_platform_popup_create(CogWlViewport *viewport, WebKitOptionMenu *option_menu)
 {
+    CogWlPlatform *platform = (CogWlPlatform *) cog_platform_get();
     g_assert(!platform->popup);
 
-    CogWlPopup *popup = cog_wl_popup_create(platform, option_menu);
+    CogWlPopup *popup = cog_wl_popup_create(viewport, option_menu);
     platform->popup = popup;
 }
 
 void
-cog_wl_platform_popup_destroy(CogWlPlatform *platform)
+cog_wl_platform_popup_destroy()
 {
+    CogWlPlatform *platform = (CogWlPlatform *) cog_platform_get();
+
     g_assert(platform->popup);
     g_clear_pointer(&platform->popup, cog_wl_popup_destroy);
 }
 
 void
-cog_wl_platform_popup_update(CogWlPlatform *platform)
+cog_wl_platform_popup_update()
```

Comment:
Same here, missing `void`:

```suggestion
cog_wl_platform_popup_update(void)
```